hypershift-operator/main.go

```diff
@@ -416,15 +418,23 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 	}
 
 	// If enabled, start controller to ensure UWM stack is enabled and configured
-	// to remotely write telemetry metrics
-	if opts.EnableUWMTelemetryRemoteWrite {
+	// to remotely write telemetry metrics. The UWM stack will be disabled if the
+	// cluster is IPv6.
+	mgmtNetworkConfig := manifests.OpenshiftNetworkConfiguration()
+	if err := mgr.GetClient().Get(ctx, crclient.ObjectKeyFromObject(mgmtNetworkConfig), mgmtNetworkConfig); err != nil {
+		return fmt.Errorf("failed to get network config resource: %w", err)
+	}
+	if opts.EnableUWMTelemetryRemoteWrite && !utilsnet.IsIPv6String(mgmtNetworkConfig.Spec.ServiceNetwork[0]) {
```

Comment:
what's the error that this would though in disconnected atm? it's not clear by reading the ticket.

Reply:
Basically the HO pod get blocked in the reconciliation and the log shows:

```
{"level":"error","ts":"2023-12-20T15:23:01Z","msg":"Reconciler error","controller":"deployment","controllerGroup":"apps","controllerKind":"Deployment","Deployment":{"name":"operator","namespace":"hypershift"},"namespace":"hypershift","name":"operator","reconcileID":"451fde3c-eb1b-4cf0-98cb-ad0f8c6a6288","error":"cannot get telemeter client secret: Secret \"telemeter-client\" not found","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227"}

{"level":"debug","ts":"2023-12-20T15:23:01Z","logger":"events","msg":"Failed to ensure UWM telemetry remote write: cannot get telemeter client secret: Secret \"telemeter-client\" not found","type":"Warning","object":{"kind":"Deployment","namespace":"hypershift","name":"operator","uid":"c6628a3c-a597-4e32-875a-f5704da2bdbb","apiVersion":"apps/v1","resourceVersion":"4091099"},"reason":"ReconcileError"}
```

It's intrusive in the sense of that block the HC reconciliation not just the normal functioning of the operator.

Reply:
>Basically the HO pod get blocked in the reconciliation

Can you clarify why? The error you shared is from a different controller, it shouldn't block HC reconciliation

The error you shared says it can't get a secret, how is that related to disconnected?

>Because we are assuming that if the first entry of the serviceNetwork in the MGMT cluster is IPv6, the HostedClusters are disconnected

I don't think that's a fair assumption. If --enable-uwm-telemetry-remote-write is enabled and telemetry is not reached because it's disconnected we should just log it.